main.cpp

```diff
@@ -82,11 +159,29 @@ struct TopRopeRoute
      int height = 15;
      int routes = 20;
 
-    void mountainFeatures( TopRopeRoute face, BoulderProblem base );
+    void mountainFeatures( TopRopeRoute face, BoulderProblem base, double mountain );
 
-    void constructMountain( int height, int routes );
+    void constructMountain( double baseDiameter );
  };
 
+void Mountain::mountainFeatures(TopRopeRoute face, BoulderProblem base, double mountain )
+{
+       for ( int i = this->routes; i > 0; i-- ) // not sure if I'm using pointers right
+       {
+            face.buildRoute( this->height, ( mountain + base.calculateDifficulty( 20.25 ) ) );
+       }
```

Comment:
Pointers aren't introduced until chapter 3.   but nice work using `this` correctly.